exes/tfcctl/src/tfcctl.cpp

```diff
@@ -76,6 +76,25 @@ inline auto stdin_coro(asio::io_context& ctx, std::string_view signal_name) -> a
   }
 }
 
+template <typename slot_type>
+void create_and_run_slot(asio::io_context& ctx,
+                         tfc::ipc_ruler::ipc_manager_client& client,
+                         const std::string& slot,
+                         const std::string& description) {
+  slot_type slot_{ ctx, client, slot, description, [](typename slot_type::value_t new_value) {
+                    if constexpr (tfc::stx::is_expected<std::remove_cvref_t<decltype(new_value)>>) {
+                      if (new_value.has_value()) {
+                        fmt::println("New value on slot: {}", new_value.value());
+                      } else {
+                        fmt::println("Error on slot: {}", new_value.error());
+                      }
+                    } else {
+                      fmt::println("New value on slot: {}", new_value);
+                    }
+                  } };
+  ctx.run();
```

Comment:
This is not a good placement for running the context.

Reply:
the problem is the scope for the slot, i cannot declare the slot before deducing the type because of the constructor, declaring a variant also doesn't work.

Reply:
next commit does have an alterior solution using optionals, take a look at that

Reply:
The current solution is the only one that  I have found to work

Reply:
You need to better structure the lifetimes expressed here. Lets look better into this before merging. You can still use this version for local testing.

Reply:
> the problem is the scope for the slot, i cannot declare the slot before deducing the type because of the constructor, declaring a variant also doesn't work.

How does the variant fail? can this not be used: https://github.com/Skaginn3x/framework/blob/main/libs/ipc/inc/public/tfc/ipc.hpp#L216